etna/ciim/models.py

```diff
@@ -181,9 +163,6 @@ def count(self):
         self._fetch(start=0, size=0)
         return self._response["hits"]["total"]["value"]
 
-    def __iter__(self):
-        return SearchQueryIterator(self)
-
     def first(self):
         """Return the first item in resultset without or None if nothing is found."""
         try:
```

Comment:
Not sure why do we rely on exceptions to return, can we do something like this if self[0] is None: return None

Reply:
Unlike a lot of languages where an exception needs to be handled in an _exceptional_ case, in Python it's common to first attempt to perform an action (like fetching an item by it's key) and handle the fact possibility of the value not being present than first checking for it:

Easier to ask for forgiveness than permission ([EAFP](https://docs.python.org/3/glossary.html#term-EAFP))

vs 

Look before you leap ([LBYL](https://docs.python.org/3/glossary.html#term-LBYL))